gammapy/stats/prior_statistic.py

```diff
@@ -0,0 +1,21 @@
+class PriorFitStatistic:
+    """Class to compute statistics due to the priors set on models and their parameters.
+
+    Parameters
+    ----------
+    weight : float
+        Weight of the prior
+    """
+
+    def __init__(self, weight=1):
+        self.weight = weight
+
+    def stat_sum(self, models):
+        stat = 0
+        for m in models:
+            if m.prior is not None:
+                stat += m.prior_stat_sum()
+            for par in m.parameters:
+                if par.prior is not None:
+                    stat += par.prior_stat_sum()
```

Comment:
given the changes in your other PR it could be something like :
if isinstance(m, `PriorModel`):
    stat += m.evaluate()
for par in m.parameters:
                if par.prior is not None:
                    stat += par.prior.evaluate()

Reply:
Yes, it basically is the same. With the new way of handling the ``prior_stat_sum`` this ``PriorFitStatistic`` class is no longer needed. I will close the PR. Thanks @QRemy for reviewing it!